codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1098,4 +1074,58 @@ private boolean allowModifications() {
             return pendingIterations == 0;
         }
     }
+
+    /**
+     * Implementation of {@link Key} that specifies the index position of the property.
+     */
+    class PropertyKey<K, V> implements Key<K, V> {
+        private final K keyObject;
+        private final Class<V> valueType;
+        private final int index;
+
+        PropertyKey(K keyObject, Class<V> valueType, int index) {
+            this.keyObject = keyObject;
+            this.valueType = valueType;
+            this.index = index;
+        }
+
+        @Override
+        public K keyObject() {
+            return keyObject;
+        }
+    }
+
+    /**
+     * A registry of all stream property keys known by this connection.
+     */
+    private class PropertyKeyRegistry {
+        final List<PropertyKey<?, ?>> keys = new ArrayList<PropertyKey<?, ?>>(4);
```

Comment:
Not sure we need to, this is going to be used much less often.  Letting `ArrayList` do the heavy lifting seems reasonable. No?